infra/config-sync/src/main/java/com/simprints/infra/config/sync/usecase/HandleProjectStateUseCase.kt

```diff
@@ -0,0 +1,40 @@
+package com.simprints.infra.config.sync.usecase
+
+import com.simprints.infra.authlogic.AuthManager
+import com.simprints.infra.authlogic.worker.SecurityStateScheduler
+import com.simprints.infra.config.store.models.ProjectState
+import com.simprints.infra.config.sync.ProjectConfigurationScheduler
+import com.simprints.infra.eventsync.EventSyncManager
+import com.simprints.infra.images.ImageUpSyncScheduler
+import kotlinx.coroutines.flow.first
+import javax.inject.Inject
+
+internal class HandleProjectStateUseCase @Inject constructor(
+    private val configScheduler: ProjectConfigurationScheduler,
+    private val securityStateScheduler: SecurityStateScheduler,
+    private val imageUpSyncScheduler: ImageUpSyncScheduler,
+    private val eventSyncManager: EventSyncManager,
+    private val authManager: AuthManager
+) {
+
+    suspend operator fun invoke(projectId: String, state: ProjectState) {
+        if (shouldSignOut(projectId, state)) {
```

Comment:
Are you not dealing with the `PAUSED` state of the project, where you can only up-sync?

Reply:
There is a check in the event sync worker that disables down sync (it was previously checking the secure state).